react-client/src/routes/CreateTutorial.js

```diff
@@ -29,50 +34,129 @@ export default class CreateTutorial extends Component {
+    switch (type) {
+      case "create":
+        tutorialData.sections[index].sectionDetails.push(
+          sectionsDetailsDataModel
+        )
+        this.setState({
+          tutorialData
+        })
+        console.log(tutorialData, "tutorialData")
+        break
+
+      case "delete":
+        tutorialData.sections[index].sectionDetails.splice(index, 1)
+        this.setState({
+          tutorialData
+        })
+        console.log(tutorialData, "tutorialData")
+    }
   }
 
-  setTutorial() {
+  setTutorial = () => {
     console.log(this.refs)
-    this.state.tutorialData.sections.forEach(d => console.log(d))
+    // this.state.tutorialData.sections.forEach(d => console.log(d))
   }
 
   render() {
     return (
-      <div>
-        <button onClick={() => this.createTut()}>Create Tutorial</button>
-        {this.state.createTutorial && (
-          // /** flexbox on this bad boy */
-          <div>
-            <label>
-              Tutorial Name
-              <input
-                type="text"
-                name="tutorialName"
-                placeholder="Enter Tutorial Name"
-                ref="tutorialName"
-              />
-            </label>
-            <button onClick={() => this.createSection()}>Create</button>
-            {this.state.tutorialData.sections.map((data, i) => {
-              return <Section key={i} index={i} />
-            })}
-            <button type="submit" onClick={() => this.setTutorial()}>
-              Submit
-            </button>
-          </div>
-        )}
+      <div className="bg-dark">
+        <div
+          className="p-2 container"
+          style={{ minHeight: "calc(100vh - 50px)" }}
+        >
+          <Button
+            variant="primary"
+            className="m-3"
+            onClick={() => this.createTut()}
+          >
+            Create Tutorial
+          </Button>
+          {this.state.createTutorial && (
+            // /** flexbox on this bad boy */
+            <div>
+              <div className="d-flex flex-column justify-content-center align-items-center text-light">
+                <div>
+                  <label htmlFor="tutorialName" className="m-2">
+                    Tutorial Name:
+                  </label>
+                  <input
+                    id="tutorialName"
+                    className="form-control"
+                    type="text"
+                    name="tutorialName"
+                    placeholder="Enter Tutorial Name"
+                    ref="tutorialName"
+                  />
+                </div>
+                <Button
+                  variant="primary"
+                  className="m-3"
+                  onClick={e => this.createSection(e, "index", "create")}
+                >
+                  Add Section
+                </Button>
+                {this.state.tutorialData.sections.map((data, i) => {
+                  return (
+                    <>
```

Comment:
What is <> ?

Reply:
<> is just an empty placeholder tag, so that we don't want always return a div parent when rendering elements from mapping an Array.

<> can also be written as ReactFragment. 
And can be imported as such. I prefer writing <>.
const React, { Component, ReactFragment } from 'react'